plone/app/content/browser/contents/__init__.py

```diff
@@ -37,41 +35,10 @@
     Rearrange='plone.app.content.browser.content.rearrange:RearrangeOrderActionView',  # noqa
 )
 
-
-def get_top_site_from_url(context, request):
-    """Find the top-most site, which is still in the url path.
-
-    If the current context is within a subsite within a PloneSiteRoot and no
-    virtual hosting is in place, the PloneSiteRoot is returned.
-    When at the same context but in a virtual hosting environment with the
-    virtual host root pointing to the subsites, it returns the subsite instead
-    of the PloneSiteRoot.
-
-    For this given content structure:
-
-    /Plone/Subsite
-
-    It should return the following in these cases:
-
-    - Naked Plone without virtual hosting, /Plone: Plone
-    - Naked Plone without virtual hosting, /Plone/Subsite: Plone
-    - Virtual hosting which roots to the subsite: Subsite
-    """
-    url_path = urlparse(context.absolute_url()).path.split('/')
-
-    site = getSite()
-    try:
-        for idx in range(len(url_path)):
-            _path = '/'.join(url_path[:idx + 1]) or '/'
-            site_path = request.physicalPathFromURL(_path)
-            site = context.restrictedTraverse('/'.join(site_path) or '/')
-            if ISite.providedBy(site):
-                break
-    except ValueError:
-        # Refs: https://github.com/plone/plone.app.content/issues/103
-        # On error, just return getSite.
-        return getSite()
-    return site
+zope.deferredimport.deprecated(
+    "Import from Products.CMFPlone.utils instead",
+    get_top_site_from_url='Products.CMFPlone:utils.get_top_site_from_url',
```

Comment:
I'am not sure if this should read:

``get_top_site_from_url='Products.CMFPlone.utils:get_top_site_from_url',``

lets see whats the test says